cinema/views.py

```diff
@@ -2,0 +1,65 @@
+from rest_framework import viewsets
+from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession
+from cinema.serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    CinemaHallSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    MovieSessionRetrieveSerializer,
+    BaseMovieSerializer,
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.prefetch_related("genres", "actors")
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return BaseMovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            return queryset.prefetch_related("genres", "actors")
+        return queryset
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all().select_related(
+        "movie",
+        "cinema_hall"
+    )
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+        elif self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            return queryset.select_related("movie", "cinema_hall")
+        return queryset
```

Comment:
The method `get_queryset` in `MovieSessionViewSet` is missing type annotations. Adding type annotations is necessary to meet the checklist requirements for code readability and maintainability.